src/ucs/vfs/base/vfs_obj.h

```diff
@@ -9,11 +9,56 @@
 
 #include <ucs/datastruct/string_buffer.h>
 #include <ucs/sys/compiler_def.h>
+#include <stdint.h>
 
 BEGIN_C_DECLS
 
 /* This header file defines API for manipulating VFS object tree structure */
 
+
+/**
+ * Defines a function prototype to fill information about an object to a given
+ * string buffer.
+ *
+ * @param _name  Function name.
+ *
+ * @param [in]    obj      Pointer to the object.
+ * @param [inout] strb     String buffer filled with the object's information.
+ * @param [in]    arg_ptr  Optional pointer argument passed to the function
+ * @param [in]    arg_u64  Optional numeric argument passed to the function
+ */
+#define UCS_VFS_FILE_SHOW(_name) \
```

Comment:
why need this macro? Can we add just `ucs_vfs_show_primitive`?

Reply:
it's done because we need to add arg_u64 to all existing callbacks, so might as well make a macro to declare these callbacks (reduce code duplication and simplify changing the callback prototype in the future)

Reply:
imo, this makes code less readable and hard to parse by ctags/cscope/etc. Also we do not use such macro for other callback definitions (ams for instance)